### ISSUE TITLE
blas: Test Builders for Additional Architectures

### ISSUE THREAD

**Issue by jonlawlor** [gonum/blas#73]
It would be good to test goblas on architectures other than x86_64, Including various ARMs and 386.  Follows up on #64, where assembly was first introduced.

There are two possibilities that I have found: running qemu on travis (which is our current CI provider) or building custom images for shippable, which allows docker containers.

The advantages to travis is that we are already using it.  Shippable has (reportedly) faster build times, and allows you to specify a docker image, which means we could substantially increase the speed of testing (and perhaps benchmarks?) for cblas.  The downside is that we should probably stick to a single CI provider, so all of our repos would need to be changed.

I'll try putting them both together in a fork.

**Comment by jonlawlor** [comment]
This turned out to be beyond me.  There are examples of how to use qemu to chroot into an arm emulating environment on travis, which I was able to get to, but go doesn't seem to work correctly after I get there, or build from scratch.  I don't know if that is a property of travis or just a setting that I don't know about.  The shippable environment doesn't seem to support the architecture, but again, I wouldn't consider this to be a conclusive statement, only that I don't have the expertise to go forward.  If anyone else has a suggestion I would be happy to revisit this.

**Comment by fhs** [comment]
I wouldn't worry about arm yet. However, I would test on 386. You can compile Go for 386 on an amd64 system like this:

```
git clone https://go.googlesource.com/go
cd go
git checkout go1.4
cd src
GOHOSTARCH=386 ./make.bash
# ../bin/go version should say "go version go1.4 linux/386"
../bin/go test github.com/gonum/blas/...
```

**Comment by jonlawlor** [comment]
Yes I can do that.  There are also precompiled versions that should speed up the process for 386.  This should be a snap in comparison...

**Comment by jonlawlor** [comment]
386 is addressed in #96

**Comment by vladimir-ch** [comment]
Closing because this repository is no longer maintained. Development has moved to https://github.com/gonum/gonum.
